src/widgets/calendar/lv_calendar_lunar.c

```diff
@@ -0,0 +1,275 @@
+
+static const char * lunar_leep_month_name[] = {"闰正月", "闰二月", "闰三月", "闰四月", "闰五月", "闰六月",
+                                               "闰七月", "闰八月", "闰九月", "闰十月", "闰十一月", "闰腊月"
+                                              };
+
+static const char * lunar_day_name[] = {"初一", "初二", "初三", "初四", "初五",
+                                        "初六", "初七", "初八", "初九", "初十",
+                                        "十一", "十二", "十三", "十四", "十五",
+                                        "十六", "十七", "十八", "十九", "二十",
+                                        "廿一", "廿二", "廿三", "廿四", "廿五",
+                                        "廿六", "廿七", "廿八", "廿九", "三十"
+                                       };
+
+static const lv_calendar_festival_t lunar_festival[] = {
+    {"春节", 1, 1},
+    {"元宵节", 1, 15},
+    {"端午节", 5, 5},
+    {"七夕节", 7, 7},
+    {"中秋节", 8, 15},
+    {"重阳节", 9, 9},
+};
+
+static const lv_calendar_festival_t solar_festival[] = {
+    {"元旦", 1, 1},
+    {"情人节", 2, 14},
+    {"妇女节", 3, 8},
+    {"植树节", 3, 12},
+    {"消费节", 3, 15},
+    {"愚人节", 4, 1},
+    {"劳动节", 5, 1},
+    {"青年节", 5, 4},
+    {"儿童节", 6, 1},
+    {"建党节", 7, 1},
+    {"建军节", 8, 1},
+    {"教师节", 9, 10},
+    {"国庆节", 10, 1},
+    {"万圣节", 10, 31},
+    {"平安夜", 12, 24},
+    {"圣诞节", 12, 25},
+};
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_calendar_set_lunar_mode(lv_obj_t * obj, bool en)
+{
+    lv_calendar_t * calendar = (lv_calendar_t *)obj;
+    calendar->use_lunar_calendar = en;
+    lv_calendar_set_showed_date(obj, calendar->today.year, calendar->today.month);
+}
+
+const char * lv_calendar_get_day_name(lv_calendar_date_t * solar)
+{
+    uint16_t i, len;
+    lv_calendar_lunar_t lunar_date;
+    lunar_date = solar_to_lunar(solar);
+
+    if(solar->year > 2099 || solar->year < 1901)
+        return NULL;
+    len = sizeof(lunar_festival) / sizeof(lv_calendar_festival_t);
+
+    for(i = 0; i < len; i++) {
+        if((lunar_date.today.month == lunar_festival[i].month) &&
+           (lunar_date.today.day == lunar_festival[i].day) &&
+           lunar_date.leep_month == false)
+            return lunar_festival[i].festival_name;
+    }
+    len = sizeof(solar_festival) / sizeof(lv_calendar_festival_t);
+
+    for(i = 0; i < len; i++) {
+        if((solar->month == solar_festival[i].month) &&
+           (solar->day == solar_festival[i].day))
+            return solar_festival[i].festival_name;
+    }
+
+    if(lunar_date.today.day == 1) {
+        if(lunar_date.leep_month == false)
+            return lunar_month_name[lunar_date.today.month - 1];
+        else {
+            return lunar_leep_month_name[lunar_date.today.month - 1];
+        }
+    }
+
+    return (char *)lunar_day_name[lunar_date.today.day - 1];
+}
+
+/**********************
+ *  STATIC FUNCTIONS
+ **********************/
+
+static lv_calendar_lunar_t solar_to_lunar(lv_calendar_date_t * solar)
```

Comment:
I have seen application requirement to calculate lunar day, let's make it a public API `lv_calendar_solar_to_lunar`?

If the code is borrowed or improved from other source code, please give the appropriate credit.